### ISSUE TITLE
etelemetry pollutes stdout by default

### ISSUE THREAD

**Issue by effigies** [nipy/nipype#3053]
### Summary

It would be nice to be able to import nipype without having to decide between disabling etelemetry and triggering output to stdout.

```Shell
> fitlins --version
190925-23:25:59,835 nipype.utils WARNING:
	 Could not check for version updates: 
Connection to server could not be made
190925-23:25:59,836 nipype.utils INFO:
	 Running nipype version 1.2.3 (latest: Unknown)
fitlins v0.5.1+6.g4606e4f
```

I can go ahead and work to delay imports so that running `--help` or `--version` never hits `import nipype`, but that seems a bit much to ask of most tools.

Perhaps we could make it trigger on something like on the first instantiation of a workflow or interface, instead of import?

### Execution environment

- `circleci/python:latest`

**Comment by satra** [comment]
@effigies - i agree - this is something i was trying to help with by encapsulating the etelemetry into a check in #3049 

so we can definitely move the check as a one time check in either `EngineBase` or `BaseInterface`, or perhaps just `BaseInterface`, since that will be used for any workflow node. but i think importing nipype also imports function, which would then trigger this baseinterface check. 

the alternative is to make this happen with the first run call on an interface or workflow. that may be a better situation.

**Comment by effigies** [comment]
I think you can do it in `EngineBase.__init__` without triggering on import. I just added a `print` statement:

```Python
>>> import nipype
>>> _ = nipype.pipeline.engine.base.EngineBase(name="test")                                                                           
Initialized
```

What if we decorate `check_version` with `@functools.lru_cache` so we don't need to maintain a global state? Then we can just update `EngineBase.__init__` with very simple logic:

```Python
class EngineBase(object):
    """Defines common attributes and functions for workflows and nodes."""

    def __init__(self, name=None, base_dir=None):
        """ Initialize base parameters of a workflow or node

        Parameters
        ----------
        name : string (mandatory)
            Name of this node. Name must be alphanumeric and not contain any
            special characters (e.g., '.', '@').
        base_dir : string
            base output directory (will be hashed before creations)
            default=None, which results in the use of mkdtemp

        """
        self._hierarchy = None
        self.name = name
        self._id = self.name # for compatibility with node expansion using iterables

        self.base_dir = base_dir
        self.config = deepcopy(config._sections)
        if self.config.getboolean('execution', 'check_version'):
            check_version()
```

**Comment by oesteban** [comment]
I was about to report this. Thanks!

**Comment by effigies** [comment]
Proposed satra#30. Not sure where we want to continue this discussion, so now all three threads are linked.

**Comment by effigies** [comment]
Fixed in #3049
